django_test_migrations/sql.py

```diff
@@ -0,0 +1,108 @@
+# -*- coding: utf-8 -*-
+
+from functools import partial
+from typing import Callable, Dict, List, Optional, Union
+
+from django.core.management.color import Style, no_style
+from django.db import DefaultConnectionProxy, connections, transaction
+from django.db.backends.base.base import BaseDatabaseWrapper
+
+_Connection = Union[DefaultConnectionProxy, BaseDatabaseWrapper]
+
+DJANGO_MIGRATIONS_TABLE_NAME = 'django_migrations'
+
+
+def drop_models_tables(
+    database_name: str,
+    style: Optional[Style] = None,
+) -> None:
+    """Drop all installed Django's models tables."""
+    style = style or no_style()
+    connection = connections[database_name]
+    tables = connection.introspection.django_table_names(
+        only_existing=True,
+        include_views=False,
+    )
+    sql_drop_tables = [
+        connection.SchemaEditorClass.sql_delete_table % {
+            'table': style.SQL_FIELD(connection.ops.quote_name(table)),
+        }
+        for table in tables
+    ]
+    if sql_drop_tables:
+        get_execute_sql_flush_for(connection)(database_name, sql_drop_tables)
+
+
+def flush_django_migrations_table(
+    database_name: str,
+    style: Optional[Style] = None,
+) -> None:
+    """Flush `django_migrations` table."""
+    style = style or no_style()
+    connection = connections[database_name]
+    django_migrations_sequences = get_django_migrations_table_sequences(
+        connection,
+    )
+    execute_sql_flush = get_execute_sql_flush_for(connection)
+    execute_sql_flush(
+        database_name,
+        connection.ops.sql_flush(
+            style,
+            [DJANGO_MIGRATIONS_TABLE_NAME],
+            django_migrations_sequences,
+            allow_cascade=False,
+        ),
+    )
+
+
+def get_django_migrations_table_sequences(
+    connection: _Connection,
+) -> List[Dict[str, str]]:  # pragma: no cover
+    """`django_migrations` table introspected sequences.
+
+    Returns properly inspected sequences when using `Django>1.11`
+    and static sequence for `id` column otherwise.
+
+    """
+    if hasattr(connection.introspection, 'get_sequences'):  # noqa: WPS421
```

Comment:
Maybe we can refactor this into `except AttributeError`? Or do you like this solution better?

Reply:
Usually I prefer `try` `except`, because I am fan of "ask for forgiveness, not permission" approach, however in this case I think usage of `hasattr()` is more elegant, because it doesn't cause broad `try except` or creating new function, however I can refactor it you prefer `try except`